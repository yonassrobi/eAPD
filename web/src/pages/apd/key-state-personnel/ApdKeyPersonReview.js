import PropTypes from 'prop-types';
import React, { Fragment, useMemo } from 'react';
import Dollars from '../../../components/Dollars';
import Review from '../../../components/Review';

const ApdStateKeyPerson = ({
  expand,
  index,
  item: { costs, email, hasCosts, name, fte, position, split, medicaidShare },
  onDeleteClick,
  apdType
}) => {
  const primary = index === 0;
  let displayName = name;
  if (name === '') {
    displayName = primary
      ? 'Primary Point of Contact name not specified'
      : 'Key Personnel name not specified';
  }

  const costByYear = useMemo(
    () => (
      <div className="ds-u-margin-top--2">
        {hasCosts === true ? (
          Object.keys(costs).map(year => (
            <div key={year} className="ds-u-padding-bottom--2">
              <strong>FFY {year} Cost: </strong>
              <Dollars>{costs[year]}</Dollars> | <strong>FTE: </strong>
              {fte[year]} | <strong>Total: </strong>
              <Dollars>{costs[year] * fte[year]} </Dollars>
              {apdType === 'HITECH' && (
                <div>
                  <strong>Federal Share: </strong>
                  <Dollars>
                    {costs[year] * fte[year] * (split[year].federal / 100)}
                  </Dollars>
                </div>
              )}
              {apdType === 'MMIS' && (
                <div>
                  <strong>Total Computable Medicaid: </strong>
                  <Dollars>
                    {costs[year] * fte[year] * (medicaidShare[year] / 100)}
                  </Dollars>{' '}
                  ({medicaidShare[year]}% Medicaid Share) |{' '}
                  <strong>Federal Share: </strong>
                  <Dollars>
                    {costs[year] *
                      fte[year] *
                      (split[year].federal / 100) *
                      (medicaidShare[year] / 100)}
                    )
                  </Dollars>
                </div>
              )}
            </div>
          ))
        ) : (
          <div>
            <strong>Total cost:</strong> <Dollars>{0}</Dollars>
          </div>
        )}
      </div>
    ),
    [hasCosts, costs, fte, split, medicaidShare, apdType]
  );

  return (
    <Fragment>
      <div className="visibility--screen">
        <Review
          heading={`${index + 1}. ${displayName}`}
          headingLevel="4"
          onDeleteClick={primary ? null : onDeleteClick}
          onEditClick={expand}
          ariaLabel={displayName}
          objType="Key Personnel"
        >
          <ul className="ds-c-list--bare">
            {primary ? <li>Primary APD Point of Contact</li> : null}
            <li>{position || 'Role not specified'}</li>
          </ul>
          {costByYear}
        </Review>
      </div>
      <div className="visibility--print">
        <Review heading={`${index + 1}. ${displayName}`}>
          <ul className="ds-c-list--bare">
            {primary ? <li>Primary APD Point of Contact</li> : null}
            <li>{position || 'Role not specified'}</li>
            <li>{email}</li>
          </ul>
          {costByYear}
        </Review>
      </div>
    </Fragment>
  );
};

ApdStateKeyPerson.propTypes = {
  expand: PropTypes.func.isRequired,
  index: PropTypes.number.isRequired,
  item: PropTypes.shape({
    costs: PropTypes.object.isRequired,
    email: PropTypes.string.isRequired,
    hasCosts: PropTypes.bool,
    name: PropTypes.string.isRequired,
    fte: PropTypes.object.isRequired,
    position: PropTypes.string.isRequired,
    split: PropTypes.object.isRequired,
    medicaidShare: PropTypes.object.isRequired
  }).isRequired,
  onDeleteClick: PropTypes.func,
  apdType: PropTypes.string.isRequired
};

ApdStateKeyPerson.defaultProps = {
  onDeleteClick: null
};

export default ApdStateKeyPerson;
