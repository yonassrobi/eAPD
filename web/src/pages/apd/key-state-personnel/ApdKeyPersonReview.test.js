import { shallow } from 'enzyme';
import React from 'react';

import ApdKeyPersonReview from './ApdKeyPersonReview';

describe('the ApdKeyPersonReview component', () => {
  const props = {
    expand: jest.fn(),
    index: 12,
    item: {
      costs: {
        1992: 100,
        1993: 300
      },
      email: 'email address',
      hasCosts: true,
      key: 'person key',
      name: 'Bob the Builder',
      fte: {
        1992: 0.32,
        1993: 0.84
      },
      split: {
        1992: {
          federal: 0,
          state: 0
        },
        1993: {
          federal: 0,
          state: 0
        }
      },
      medicaidShare: {
        1992: 100,
        1993: 100
      },
      position: 'The Builder'
    },
    onDeleteClick: jest.fn(),
    years: ['1992', '1993']
  };

  test('renders correctly for hitech apds', () => {
    const component = shallow(
      <ApdKeyPersonReview apdType={'HITECH'} {...props} />
    );
    expect(component).toMatchSnapshot();

    expect(component.find('Review').first().prop('onEditClick')).toEqual(
      props.expand
    );

    expect(component.find('Review').first().prop('onDeleteClick')).toEqual(
      props.onDeleteClick
    );
  });

  test('renders correctly for mmis apds', () => {
    const component = shallow(
      <ApdKeyPersonReview apdType={'MMIS'} {...props} />
    );
    expect(component).toMatchSnapshot();

    expect(component.find('Review').first().prop('onEditClick')).toEqual(
      props.expand
    );

    expect(component.find('Review').first().prop('onDeleteClick')).toEqual(
      props.onDeleteClick
    );
  });
});
