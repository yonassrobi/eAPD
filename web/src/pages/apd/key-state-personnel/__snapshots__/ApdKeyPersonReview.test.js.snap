// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the ApdKeyPersonReview component renders correctly for hitech apds 1`] = `
<Fragment>
  <div
    className="visibility--screen"
  >
    <Review
      ariaLabel="Bob the Builder"
      editHref={null}
      heading="13. Bob the Builder"
      headingLevel="4"
      objType="Key Personnel"
      onDeleteClick={[MockFunction]}
      onDeleteLabel={null}
      onEditClick={[MockFunction]}
      skipConfirmation={false}
      title={null}
    >
      <ul
        className="ds-c-list--bare"
      >
        <li>
          The Builder
        </li>
      </ul>
      <div
        className="ds-u-margin-top--2"
      >
        <div
          className="ds-u-padding-bottom--2"
          key="1992"
        >
          <strong>
            FFY 
            1992
             Cost: 
          </strong>
          <Dollars>
            100
          </Dollars>
           | 
          <strong>
            FTE: 
          </strong>
          0.32
           | 
          <strong>
            Total: 
          </strong>
          <Dollars>
            32
             
          </Dollars>
          <div>
            <strong>
              Federal Share: 
            </strong>
            <Dollars>
              0
            </Dollars>
          </div>
        </div>
        <div
          className="ds-u-padding-bottom--2"
          key="1993"
        >
          <strong>
            FFY 
            1993
             Cost: 
          </strong>
          <Dollars>
            300
          </Dollars>
           | 
          <strong>
            FTE: 
          </strong>
          0.84
           | 
          <strong>
            Total: 
          </strong>
          <Dollars>
            252
             
          </Dollars>
          <div>
            <strong>
              Federal Share: 
            </strong>
            <Dollars>
              0
            </Dollars>
          </div>
        </div>
      </div>
    </Review>
  </div>
  <div
    className="visibility--print"
  >
    <Review
      ariaLabel={null}
      editHref={null}
      heading="13. Bob the Builder"
      objType={null}
      onDeleteClick={null}
      onDeleteLabel={null}
      onEditClick={null}
      skipConfirmation={false}
      title={null}
    >
      <ul
        className="ds-c-list--bare"
      >
        <li>
          The Builder
        </li>
        <li>
          email address
        </li>
      </ul>
      <div
        className="ds-u-margin-top--2"
      >
        <div
          className="ds-u-padding-bottom--2"
          key="1992"
        >
          <strong>
            FFY 
            1992
             Cost: 
          </strong>
          <Dollars>
            100
          </Dollars>
           | 
          <strong>
            FTE: 
          </strong>
          0.32
           | 
          <strong>
            Total: 
          </strong>
          <Dollars>
            32
             
          </Dollars>
          <div>
            <strong>
              Federal Share: 
            </strong>
            <Dollars>
              0
            </Dollars>
          </div>
        </div>
        <div
          className="ds-u-padding-bottom--2"
          key="1993"
        >
          <strong>
            FFY 
            1993
             Cost: 
          </strong>
          <Dollars>
            300
          </Dollars>
           | 
          <strong>
            FTE: 
          </strong>
          0.84
           | 
          <strong>
            Total: 
          </strong>
          <Dollars>
            252
             
          </Dollars>
          <div>
            <strong>
              Federal Share: 
            </strong>
            <Dollars>
              0
            </Dollars>
          </div>
        </div>
      </div>
    </Review>
  </div>
</Fragment>
`;

exports[`the ApdKeyPersonReview component renders correctly for mmis apds 1`] = `
<Fragment>
  <div
    className="visibility--screen"
  >
    <Review
      ariaLabel="Bob the Builder"
      editHref={null}
      heading="13. Bob the Builder"
      headingLevel="4"
      objType="Key Personnel"
      onDeleteClick={[MockFunction]}
      onDeleteLabel={null}
      onEditClick={[MockFunction]}
      skipConfirmation={false}
      title={null}
    >
      <ul
        className="ds-c-list--bare"
      >
        <li>
          The Builder
        </li>
      </ul>
      <div
        className="ds-u-margin-top--2"
      >
        <div
          className="ds-u-padding-bottom--2"
          key="1992"
        >
          <strong>
            FFY 
            1992
             Cost: 
          </strong>
          <Dollars>
            100
          </Dollars>
           | 
          <strong>
            FTE: 
          </strong>
          0.32
           | 
          <strong>
            Total: 
          </strong>
          <Dollars>
            32
             
          </Dollars>
          <div>
            <strong>
              Total Computable Medicaid: 
            </strong>
            <Dollars>
              32
            </Dollars>
             
            (
            100
            % Medicaid Share) |
             
            <strong>
              Federal Share: 
            </strong>
            <Dollars>
              0
              )
            </Dollars>
          </div>
        </div>
        <div
          className="ds-u-padding-bottom--2"
          key="1993"
        >
          <strong>
            FFY 
            1993
             Cost: 
          </strong>
          <Dollars>
            300
          </Dollars>
           | 
          <strong>
            FTE: 
          </strong>
          0.84
           | 
          <strong>
            Total: 
          </strong>
          <Dollars>
            252
             
          </Dollars>
          <div>
            <strong>
              Total Computable Medicaid: 
            </strong>
            <Dollars>
              252
            </Dollars>
             
            (
            100
            % Medicaid Share) |
             
            <strong>
              Federal Share: 
            </strong>
            <Dollars>
              0
              )
            </Dollars>
          </div>
        </div>
      </div>
    </Review>
  </div>
  <div
    className="visibility--print"
  >
    <Review
      ariaLabel={null}
      editHref={null}
      heading="13. Bob the Builder"
      objType={null}
      onDeleteClick={null}
      onDeleteLabel={null}
      onEditClick={null}
      skipConfirmation={false}
      title={null}
    >
      <ul
        className="ds-c-list--bare"
      >
        <li>
          The Builder
        </li>
        <li>
          email address
        </li>
      </ul>
      <div
        className="ds-u-margin-top--2"
      >
        <div
          className="ds-u-padding-bottom--2"
          key="1992"
        >
          <strong>
            FFY 
            1992
             Cost: 
          </strong>
          <Dollars>
            100
          </Dollars>
           | 
          <strong>
            FTE: 
          </strong>
          0.32
           | 
          <strong>
            Total: 
          </strong>
          <Dollars>
            32
             
          </Dollars>
          <div>
            <strong>
              Total Computable Medicaid: 
            </strong>
            <Dollars>
              32
            </Dollars>
             
            (
            100
            % Medicaid Share) |
             
            <strong>
              Federal Share: 
            </strong>
            <Dollars>
              0
              )
            </Dollars>
          </div>
        </div>
        <div
          className="ds-u-padding-bottom--2"
          key="1993"
        >
          <strong>
            FFY 
            1993
             Cost: 
          </strong>
          <Dollars>
            300
          </Dollars>
           | 
          <strong>
            FTE: 
          </strong>
          0.84
           | 
          <strong>
            Total: 
          </strong>
          <Dollars>
            252
             
          </Dollars>
          <div>
            <strong>
              Total Computable Medicaid: 
            </strong>
            <Dollars>
              252
            </Dollars>
             
            (
            100
            % Medicaid Share) |
             
            <strong>
              Federal Share: 
            </strong>
            <Dollars>
              0
              )
            </Dollars>
          </div>
        </div>
      </div>
    </Review>
  </div>
</Fragment>
`;
